src/System/Integrate/Http/Karnel.php

```diff
@@ -51,6 +52,22 @@ public function handle(Request $request)
         return $pipeline($request);
     }
 
+    /**
+     * Terminate Requesr and Response.
+     */
+    public function terminate(Request $request, Response $response): void
+    {
+        $middleware = $this->dispatcherMiddleware($request) ?? [];
```

Comment:
why not directly call `Router::current()`

Reply:
this helped to test terminate function intead mock a router dispatch. i think is well to use this function as long as default is route current. so we don't need overide latter.